lib/pub_sub/poller.rb

```diff
@@ -1,31 +1,38 @@
 module PubSub
   class Poller
-    def initialize(queue_url, verbose = false)
-      @queue_url = queue_url
+    def initialize(verbose = false)
       @verbose = verbose
     end
 
     def poll
-      poller.poll(config) do |message|
-        if @verbose
-          PubSub.logger.info(
-            "PubSub received: #{message.message_id} - #{message.body}"
-          )
+      loop do
+        Breaker.current_breaker.run do
+          poller.poll(idle_timeout: 60) do |message|
+            if @verbose
+              PubSub.logger.info(
+                "PubSub received: #{message.message_id} - #{message.body}"
+              )
+            end
+            Message.new(message.body).process
+          end
         end
-        Message.new(message.body).process
+        Breaker.use_next_breaker
```

Comment:
Hmm - I thought this wouldn't be a problem, because I assumed the publisher and subscriber would be running on different threads. But now I think about it, it's entirely possible to send updates as part of a task triggered by a message.

I don't think this will cause problems in practice. Because the subscribers will (eventually) receive messages on any queue, sending messages to a random region's queue doesn't cause the behaviour to be incorrect - the message still gets received.

It has the slightly detrimental effect that ordering isn't preserved (but AWS's infrastructure doesn't guarantee ordering anyhow), and also it can mean that workers can get stuck in an inefficient position. If a message is being sent every 59 seconds, then the worker will keep listening, spending most of its time waiting, even if another queue is backed up. This is functionally the same as if the queue was getting more updates than it can handle (10 messages a second being generated, 9 being consumed). We don't guard against this (and we can't from within this gem), and the fix is identical (add more workers/computing resources).

I'll add this to the readme, and also reduce the default idle timeout to 15 seconds to reduce the chances of us running into this.